build.gradle

```diff
@@ -18,7 +18,8 @@ buildscript {
         jcenter()
     }
     dependencies {
-        classpath 'com.android.tools.build:gradle:4.0.2'
+        classpath 'com.android.tools.build:gradle:4.1.3'
+        classpath "com.vanniktech:gradle-android-junit-jacoco-plugin:0.16.0"
```

Comment:
I'm using [this library](https://github.com/vanniktech/gradle-android-junit-jacoco-plugin) instead of the official jacoco library because it currently has [this issue](https://github.com/gradle/gradle/issues/14132) :( There is an [alternative library](https://github.com/arturdm/jacoco-android-gradle-plugin/i) which codecov recommended however it has not been updated in a year and consequently also has[ compatability issues](https://github.com/arturdm/jacoco-android-gradle-plugin/issues/83)

Reply:
Thanks for the details, it was super useful. I checked the issue, and I was thinking it should not be the end of the world, because it is about the order of the task execution. A simple fix, with making the Jacoco task to depend on the connectedAndroidTask should fix it. Turned out, they made the connectedAndroidTask to depend on Jacoco task, so a single line of fix is not enough, as it would lead to circular task dependency (Jacoco task depends on connectedAndroidTest, connectedAndroidTest depends on Jacoco task).
So additionally, I had to update the task dependency of connectedAndroidTest, to not to depend on Jacoco task. Created an experimental branch for it, see the related [commit](https://github.com/bitrise-io/trace-android-sdk/commit/3b1926a547f637f655332678cfe476884e6dc1f1). The implementation of the official plugin is based on [this article](https://medium.com/wandera-engineering/android-kotlin-code-coverage-with-jacoco-sonar-and-gradle-plugin-6-x-3933ed503a6e), I just tailored it to our needs (removed some code, which are not relevant for us) and added the possible workaround/fix.

I do not want to rewrite your PR, but I was thinking we could stick with the official plugin, as I usually try to avoid 3rd party plugins. This is not a blocker, so happy to approve your PR as is, and move on, but left here for consideration (and review the experiment). 🙂